File: app/src/main/java/org/arcade/atomcity/network/TaikoServerService.java
package org.arcade.atomcity.network;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface TaikoServerService {
    @GET("PlayHistory/{{userNumber}}")
    Call<ResponseType> getUserProfile(@Path("userNumber") String userNumber);

    @GET("GameData/MusicDetails")
    Call<ResponseType> getMusicDetails();
}

File: app/src/main/java/org/arcade/atomcity/network/TaikoServerService.kt
package org.arcade.atomcity.network;

import org.arcade.atomcity.model.taikoserver.usersettings.TaikoServerUserSettingsResponse
import org.arcade.atomcity.model.taikoserver.musicDetails.TaikoServerMusicDetailsResponse;
import org.arcade.atomcity.model.taikoserver.songHistory.TaikoServerPlayHistoryResponse;

import retrofit2.http.GET;
import retrofit2.http.Path;

interface TaikoServerService {
    @GET("PlayHistory/{userNumber}")
    suspend fun getPlayHistory(
        @Path("userNumber") userNumber: String
    ): TaikoServerPlayHistoryResponse

    @GET("GameData/MusicDetails")
    suspend fun getMusicDetails()
    : TaikoServerMusicDetailsResponse

    @GET("UserSettings/{userNumber}")
    suspend fun getUserSettings(
        @Path("userNumber") userNumber: String
    ): TaikoServerUserSettingsResponse

    @GET("PlayData/152")
    suspend fun getPlayData(): String
}
